http/Router_utils.cpp

```diff
@@ -135,28 +149,63 @@ void Router::validateContentType() {
     }
 }
 
-void Router::setParsedURL(void) {
-    const std::string& URL = request.getURL();
-
-    //WIP: get real path from server.conf
-    parsedURL = URL; //temporary
+void Router::setConfigURL() {
+    std::string&        URLFromRequest = const_cast<std::string&>(request.getURL());
+    std::string         bestMatchURL;
+    std::string         bestMatchRoot;
+    Config::location    bestLocation;
+    
+    GetBestMatchURL(config->locations, URLFromRequest, bestMatchURL, bestMatchRoot, bestLocation);
+    if (bestMatchURL == "/" && URLFromRequest == "/") {
```

Comment:
해당 if문에 bestMatchURL == "/" 인 상황에 들어오게 되는데, 161 라인에서 bestMatchURL 가 비어있는 경우에 들어가게 되서 절대로 들어갈 수 없는 조건문이 쓰여있어요!

Reply:
161 line -> `if (bestMatchRoot.empty()) { `이렇게 수정하면 됌

Reply:
````
void Router::setConfigURL() {
    std::string&        URLFromRequest = const_cast<std::string&>(request.getURL());
    std::string         bestMatchURL;
    std::string         bestMatchRoot;
    Config::location    bestLocation;
    
    getBestMatchURL(config->locations, URLFromRequest, bestMatchURL, bestMatchRoot, bestLocation);
    // when client ask directory 
    if (URLFromRequest.back() == '/') {
        URLFromRequest.resize(URLFromRequest.length() - 1);
        // no root in location , no matching with any location 
        if (bestMatchRoot.empty()) {
            configURL = findPotentialIndexPath(config->root, config->index, URLFromRequest);
            configRoot = config->root;
        // matching with location 
        } else {
            configURL = findPotentialIndexPath(bestMatchRoot, bestLocation.index, URLFromRequest);
            configRoot = bestMatchRoot;
        }
        // add function about autoindex in case there's no index exists. 
    } else {
      if (bestMatchURL.empty() || bestMatchRoot.empty()) {
          configURL = findPath(config->root, URLFromRequest);
          configRoot = config->root;
      } else {
          configURL = findPath(bestMatchRoot, URLFromRequest);
          configRoot = bestMatchRoot + bestMatchURL;
      }
   }
}
````

Reply:
이런식으로 수정하면 될듯합니다! 아주 깔끔한 코드에요. 
처음 나오는 if 문이 디렉토리를 요구한 경우 전체에 해당하고, 
각각 server블록 정보를 사용하는 경우와 location 블록 정보를 사용하는 부분으로 쪼개어 집니다.

Reply:
if (bestMatchURL == "/" && URLFromRequest == "/") {
        URLFromRequest = "";
        if (bestMatchURL.empty() || bestMatchRoot.empty()) {
            
이 부분이 

    if (URLFromRequest.back() == '/') {
        URLFromRequest.resize(URLFromRequest.length() - 1);
        if (bestMatchRoot.empty()) {
        
      이렇게 바뀌었는데,         if (bestMatchRoot.empty()) { 이 부분은 써주신 것처럼 조건을 하나 줄이는 방식으로 수정했습니다!

          if (URLFromRequest.back() == '/') {
        URLFromRequest.resize(URLFromRequest.length() - 1);
        이 부분의 경우에는 "/"인 경우와 일반 디렉토리인 경우 다르게 판단을 해 줘야 할 것 같아서 함수를 새로 만들어줬습니다. 

다만, 구체적인 로직은 이전에 쓰던 것을 그대로 쓸 수는 없을 것 같습니다. 원래 디렉토리가 들어오는 경우 에러일 거라고 가정하고 코드를 짰던 건데, 디렉토리일 경우 따로 판단을 해 줘야 한다면 bestMatchRoot로 돌리는 것은 이상하기 때문입니다 ㅎㅎ